src/sdk/pynni/nni/msg_dispatcher.py

```diff
@@ -144,7 +145,10 @@ def handle_report_metric_data(self, data):
             assert data['trial_job_id'] is not None
             assert data['parameter_index'] is not None
             param_id = _create_parameter_id()
-            param = self.tuner.generate_parameters(param_id, trial_job_id=data['trial_job_id'])
+            try:
+                param = self.tuner.generate_parameters(param_id, trial_job_id=data['trial_job_id'])
+            except NoMoreTrialError:
+                param = None
```

Comment:
i am thinking about that whether we should mention this logic in doc?

Reply:
This is stated in MultiPhase.md   currently the api document is generated automatically, there is no much content in it.